bootstrap-agent-api/src/lib.rs

```diff
@@ -0,0 +1,92 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! The bootstrap agent's API.
+//!
+//! Note that the bootstrap agent also communicates over Sprockets,
+//! and has a separate interface for establishing the trust quorum.
+
+use dropshot::{
+    HttpError, HttpResponseOk, HttpResponseUpdatedNoContent, RequestContext,
+    TypedBody,
+};
+use omicron_common::api::external::SemverVersion;
+use omicron_uuid_kinds::{RackInitUuid, RackResetUuid};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use sled_agent_types::{
+    rack_init::RackInitializeRequest, rack_ops::RackOperationStatus,
+};
+use sled_hardware_types::Baseboard;
+
+#[dropshot::api_description {
+    module = "bootstrap_agent_api_mod",
+}]
+pub trait BootstrapAgentApi {
+    type Context;
+
+    /// Return the baseboard identity of this sled.
+    #[endpoint {
+        method = GET,
+        path = "/baseboard",
+    }]
+    async fn baseboard_get(
+        rqctx: RequestContext<Self::Context>,
+    ) -> Result<HttpResponseOk<Baseboard>, HttpError>;
+
+    /// Provide a list of components known to the bootstrap agent.
```

Comment:
I changed these to be "Provide" instead of "Provides" etc, because that's the general style we use within Omicron. (I personally prefer "Provides" in my own projects, but I think consistency is more important.)